packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -2188,6 +2193,89 @@ export class ProjectService {
         return charts;
     }
 
+    async getMostPopularAndRecentlyUpdated(
+        user: SessionUser,
+        projectUuid: string,
+    ): Promise<MostPopularAndRecentlyUpdated> {
+        if (
+            user.ability.cannot(
+                'view',
+                subject('Project', {
+                    organizationUuid: user.organizationUuid,
+                    projectUuid,
+                }),
+            )
+        ) {
+            throw new ForbiddenError();
+        }
+
+        const spaces = await this.spaceModel.find({ projectUuid });
+        const allowedSpaces = spaces.filter(
+            (space) =>
+                space.projectUuid === projectUuid &&
+                hasSpaceAccess(user, space, true),
+        );
+
+        const mostPopular = await this.getMostPopular(allowedSpaces);
+        const recentlyUpdated = await this.getRecentlyUpdated(allowedSpaces);
+
+        return {
+            mostPopular: mostPopular
+                .sort((a, b) => b.views - a.views)
+                .slice(
+                    0,
+                    this.spaceModel.MOST_POPULAR_OR_RECENTLY_UPDATED_LIMIT,
```

Comment:
is this slice limit necessary ? we're already limiting the results on the query

Reply:
Oh yes, this combines the most popular dashboards and most popular charts, so they might be 20. But we resort by views, and then slice by the max: `10`